runtime/lua/vim/lsp/util.lua

```diff
@@ -297,6 +297,11 @@ local function get_lines(bufnr, rows)
 
   local filename = api.nvim_buf_get_name(bufnr)
 
+  if vim.fn.isdirectory(filename) then
+    vim.fn.bufload(bufnr)
+    return buf_lines()
+  end
```

Comment:
Alternatively, could handle the `fs_read` result below. Should it silently return empty then? I don't fully understand what this function does.